compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -315,7 +315,10 @@ object Checking {
         fail(AbstractMemberMayNotHaveModifier(sym, flag))
     def checkNoConflict(flag1: FlagSet, flag2: FlagSet) =
       if (sym.is(allOf(flag1, flag2)))
-        fail(i"illegal combination of modifiers: $flag1 and $flag2 for: $sym")
+        fail(i"illegal combination of modifiers: `$flag1` and `$flag2` for: $sym")
+    def checkApplicable(flag: FlagSet, ok: Boolean) =
+      if (!ok && !sym.is(Synthetic))
```

Comment:
Why is the `!sym.is(Synthetic)` needed here? Is it only for the `lazy val` of the modules?